src/twisted/internet/tcp.py

```diff
@@ -334,10 +334,16 @@ def logPrefix(self):
         return self.logstr
 
     def getTcpNoDelay(self):
-        return bool(self.socket.getsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY))
+        try:
```

Comment:
I think the TCP transport implementation should be left as it is

Many thanks for the report.

We have the [ITCPTransport](https://docs.twisted.org/en/latest/api/twisted.internet.interfaces.ITCPTransport.html) interface that defined the setTcpNoDelay

I think that in the HTTP code we need to conditionally call `self.transport.setTcpNoDelay(True)` only if the transport implements the `ITCPTransport` interface

Reply:
It already does, pretty sure, the issue is that Unix transport pretends to be TCP transport?

Reply:
True. This comment was before doing more troubleshooting